### PR TITLE
[SDPA-1636] adds tide_media_file_overwrite submodule

### PR DESCRIPTION
**Pull request by vincent-gao**
Jira: https://digital-engagement.atlassian.net/browse/SDPA-1636

### Changes
1. code changes
2. bug fixes
3. made tide_media_file_overwrite to be a submodule of tide_media
4. override behat.yml coming from dev-tools

relates to https://github.com/dpc-sdp/content-vic-gov-au/pull/648

### CODE REVIEW

**Comment by vincent-gao** [comment]
test failed, because of clamav module that doesn't get enabled in PR env

File: modules/tide_media_file_overwrite/tide_media_file_overwrite.module [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+/**
+ * Implements hook_help().
+ */
+function tide_media_file_overwrite_help($route_name, RouteMatchInterface $route_match) {
+  switch ($route_name) {
+    // Main module help for the tide_media_file_overwrite module.
+    case 'help.page.tide_media_file_overwrite':
+      $output = '';
+      $output .= '<h3>' . t('About') . '</h3>';
+      $output .= '<p>' . t('Gives the author option to overwrite file, and keeping the same filename when upload.') . '</p>';
```
Comment (kurtfoster) [inline modules/tide_media_file_overwrite/tide_media_file_overwrite.module]:
```suggestion
      $output .= '<p>' . t('Gives the author option to a overwrite file, and keeping the same filename when upload. This is an override of the default Drupal behaviour where a file is uploaded as s new file with a number appened to the file name, eg my-file-title_1.pdf'). This module can be set to Override of not Override the file by default. Each file upload has an option to change the default.'</p>';
```

Comment (vincent-gao) [reply modules/tide_media_file_overwrite/tide_media_file_overwrite.module]: applied.

File: modules/tide_media_file_overwrite/tests/behat/features/configuration.feature [hunk 2]
```diff
@@ -0,0 +1,15 @@
+@tide @skipped
```
[the remark this answers is not in the document]
Comment (vincent-gao) [reply modules/tide_media_file_overwrite/tests/behat/features/configuration.feature]: Thanks @GROwen . removed it.

File: modules/tide_media_file_overwrite/tests/behat/bootstrap/TideMediaFileOverwriteContext.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?php
```
[the remark this answers is not in the document]
Comment (vincent-gao) [reply modules/tide_media_file_overwrite/tests/behat/bootstrap/TideMediaFileOverwriteContext.php]: Thanks @GROwen . deleted it.

**Comment by vincent-gao** [comment]
Currently, there is no way to fix the test failed. we need clamav service in CI env. I will follow up with @steveworley

File: composer.json [hunk 4]
```diff
@@ -4,7 +4,7 @@
     "type": "drupal-module",
     "license": "GPL-2.0-or-later",
     "require": {
-        "dpc-sdp/tide_core": "^1.2.6",
+        "dpc-sdp/tide_core": "dev-feature/SDPA-3505-turn-off-clamav-module-in-tide_core_module as 1.4.3",
```
Comment (kurtfoster) [inline composer.json]: @vincent-gao , do you still need to do this, this PR is merged?

Comment (vincent-gao) [reply composer.json]: @kurtfoster , this `dev-feature/SDPA-3505-turn-off-clamav-module-in-tide_core_module` will not be merged. I will release a new tide_core to be added.

**Pull request merged**